### DISCUSSION TITLE
Best way to set up a class of students with Jupyter notebooks in Github Education

### DISCUSSION THREAD

**Discussion by heemlanden-bch** [community/community#31271] [GitHub Education]
I'm teaching computer science to high school students, and looking for a way to set them up with Jupyter notebooks in Python and R without having to debug installations in their BYOD laptops. I'm aware of Google Colaboratory but since my school uses Microsoft accounts, I am looking for an equivalent in the Microsoft ecosystem so I don't have to ask them to create Google accounts just for my course.

From the documentation, it looks like either Azure ML Studio or Github Codespaces could be a way to get there, but is there anyone that has experience with this? I have a verified Github Education account but don't have access to Codespaces in the organizations I upgraded to Github Team.

In the ideal situation, I could create an assignment repository with a number of .ipynb notebooks, and the students would clone those (e.g. via Github Education classroom), be able to edit these in an interactive online Jupyter environment, and submit their changes via a PR once they are done with their assignment.

**Comment by langdon** [comment]
I am doing exactly this for the first time this semester. I have created a classroom, added all the students. I have created template repos for labs, homeworks and lectures. I then make assignments of all 3 types. I distribute the invite link. I also have the students change their codespace settings to default to jupyterlab. I then have the students follow the invite link, code button->codespace, make their mods, download the notebook, upload it to gradescope. I have not experimented with direct github repo->gradescope because I assume the repos all being private will cause issues.

*Strongly* recommend creating a new org for the instance of the class as you will get a repo for every student * every assignment in the org. I also found putting a private repo template for the assignment directly in the org was better than having them separated.

Still experimenting and I just filed this bug: https://github.com/orgs/community/discussions/67130.

But let me know how it goes for you and/or if you want to see how I did it. Apologies that the first part is a bit disjointed. Hard to describe a workflow that is a bunch of parallel steps :).
